### PR TITLE
feat(deployments): add legend to donut charts

### PR DESCRIPTION
**Pull request by jiekang**
This addresses https://github.com/openshiftio/openshift.io/issues/1534

### CODE REVIEW

**Comment by andrewazores** [comment]
Could you share a screenshot?

**Comment by jiekang** [comment]
Preview:

donut-legend

**Comment by andrewazores** [comment]
Thanks, looks good visually.

I think we should wait for Adam's review comments on your other PR (https://github.com/fabric8-ui/fabric8-ui/pull/2323) and adapt this one to suit if necessary.

**Comment by fabric8cd** [comment]
@jiekang fabric8/fabric8-ui:SNAPSHOT-PR-2333-3 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2333-fabric8-ui.badger.fabric8.io

File: src/app/space/create/deployments/deployments-donut/deployments-donut-chart/deployments-donut-chart.component.spec.ts [hunk 1]
```diff
@@ -22,6 +27,18 @@ describe('DeploymentsDonutChartComponent', () => {
     pods = Observable.of({ pods: [['Running', 1], ['Terminating', 1]], total: 2 });
     desiredReplicas = 1;
     isIdled = false;
+    colors = {
```
Comment (mindreeper2420) [inline src/app/space/create/deployments/deployments-donut/deployments-donut-chart/deployments-donut-chart.component.spec.ts]: Please use the PatternFly colors for all of these items and comment what variables they are. Example:
```
'Running': '#004368', // @color-pf-blue-600
```

File: src/app/space/create/deployments/deployments-donut/deployments-donut.component.ts [hunk 2]
```diff
@@ -24,6 +24,18 @@ export class DeploymentsDonutComponent implements OnInit {
   desiredReplicas: number;
   debounceScale = debounce(this.scale, 650);
 
+  colors = {
```
Comment (mindreeper2420) [inline src/app/space/create/deployments/deployments-donut/deployments-donut.component.ts]: Please use the PatternFly colors for all of these items and comment what variables they are. Example:
```
'Running': '#004368', // @color-pf-blue-600
```

File: src/app/space/create/deployments/deployments-donut/deployments-donut.component.less [hunk 3]
```diff
@@ -1,11 +1,33 @@
 @import (reference) '../../../../../assets/stylesheets/shared/osio.less';
 
+.deployments-donut {
```
Comment (mindreeper2420) [inline src/app/space/create/deployments/deployments-donut/deployments-donut.component.less]: classes should be stacked. Example:
```
.deployments {
  &-donut {
    &-legend {
    }
    &-square {
    }
    &-scale-controls {
      a {
      }
    }
  }
}
```

**Comment by fabric8cd** [comment]
@jiekang fabric8/fabric8-ui:SNAPSHOT-PR-2333-4 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2333-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@jiekang fabric8/fabric8-ui:SNAPSHOT-PR-2333-5 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2333-fabric8-ui.badger.fabric8.io

File: src/app/space/create/deployments/deployments-donut/deployments-donut.component.less [hunk 4]
```diff
@@ -1,39 +1,63 @@
...
+    &-scale-controls {
+      -webkit-box-pack: center;
+      justify-content: center;
+      font-size: 24px;
+      display: inline-block;
+      vertical-align: middle;
+      margin-right: 10px;
+      margin-left: 10px;
+      a {
+        color: #bbb;
```
Comment (mindreeper2420) [inline src/app/space/create/deployments/deployments-donut/deployments-donut.component.less]: this should be a color variable

File: src/app/space/create/deployments/deployments-donut/deployments-donut.component.less [hunk 5]
```diff
@@ -1,39 +1,63 @@
...
+        &:hover,
+        &:active {
+          color: #72767b;
```
Comment (mindreeper2420) [inline src/app/space/create/deployments/deployments-donut/deployments-donut.component.less]: color variable

File: src/app/space/create/deployments/deployments-donut/deployments-donut.component.less [hunk 6]
```diff
@@ -1,39 +1,63 @@
...
+          text-decoration: none;
+        }
+        &.disabled {
+          opacity: .4;
+          cursor: not-allowed;
+          &:hover {
+            color: #bbb;
```
Comment (mindreeper2420) [inline src/app/space/create/deployments/deployments-donut/deployments-donut.component.less]: color variable

File: src/app/space/create/deployments/deployments-donut/deployments-donut-chart/deployments-donut-chart.component.less [hunk 7]
```diff
@@ -3,34 +3,38 @@
...
+  .c3-tooltip-container {
+    top: -27px;
+    left: 50%;
+    -webkit-transform: translateX(-50%);
+            transform: translateX(-50%);
+    white-space: nowrap;
+  }
+
+  path.c3-arc-Empty {
+    stroke: #d1d1d1;
```
Comment (mindreeper2420) [inline src/app/space/create/deployments/deployments-donut/deployments-donut-chart/deployments-donut-chart.component.less]: please update to a color variable

File: src/app/space/create/deployments/deployments-donut/deployments-donut-chart/deployments-donut-chart.component.less [hunk 8]
```diff
@@ -3,34 +3,38 @@
...
+  &.mini {
+    display: inline-block;
+    margin: 0 0 0 -8px;
+    min-width: 45px;
+    min-height: 45px;
+    top: 3px;
+    vertical-align: middle;
+  }
+
+  .c3-defocused {
```
Comment (mindreeper2420) [inline src/app/space/create/deployments/deployments-donut/deployments-donut-chart/deployments-donut-chart.component.less]: you can stack all of the `.c3-` classes here as well

**Comment by fabric8cd** [comment]
@jiekang fabric8/fabric8-ui:SNAPSHOT-PR-2333-6 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2333-fabric8-ui.badger.fabric8.io

**Pull request merged**